scene/gui/code_edit.cpp

```diff
@@ -742,6 +742,48 @@ void CodeEdit::_backspace_internal(int p_caret) {
 	end_complex_operation();
 }
 
+/* Folding Block*/
+bool CodeEdit::_is_fold_blocks_start(int p_line) const {
+	ERR_FAIL_INDEX_V(p_line, get_line_count(), false);
+	for (int i = 0; i < fold_blocks_prefixes.size(); i++) {
+		// Empty string prefix will always return true when begins_with is checked
+		if (fold_blocks_prefixes[i] == "") {
+			continue;
+		}
+		// this line and next  but not previous begins with fold block prefix
+		if (get_line(p_line).begins_with(fold_blocks_prefixes[i]) && get_line(p_line + 1).begins_with(fold_blocks_prefixes[i])) {
+			if (get_line(p_line - 1).begins_with(fold_blocks_prefixes[i])) {
+				return false;
+			}
+			return true;
+		}
+	}
```

Comment:
```suggestion
	for (const String &prefix : fold_blocks_prefixes) {
		// Empty string prefix will always return true when begins_with is checked.
		if (prefix.is_empty()) {
			continue;
		}
		// This line and next, but not previous, begins with fold block prefix.
		if (get_line(p_line).begins_with(prefix) && get_line(p_line + 1).begins_with(prefix)) {
			return !get_line(p_line - 1).begins_with(prefix);
		}
	}
```
Some simplifications and style details